packages/@sanity/form-builder/src/inputs/ObjectInput/ObjectInput.tsx

```diff
@@ -99,13 +99,14 @@ export const ObjectInput = memo(
 
     const renderField = React.useCallback(
       (field: ObjectField, fieldLevel: number, index: number) => {
-        if (!filterField(type, field) || field.type.hidden) {
+        const fieldValue = value?.[field.name]
+        if (!filterField(type, field)) {
           return null
         }
 
-        const fieldValue = value && value[field.name]
         return (
           <ObjectInputField
+            parent={value}
```

Comment:
I think it would be better if you wrapped the parent value in a ref and synced the ref via `useLayoutEffect`. this will create a stable value that won't cause re-renders and remove the need for the useMemo introduced in `ObjectInputField`.

it should be fine to have parent in a ref since the condition callback function has the ability to pull values

```js
function ObjectInput({value, /* ... */}) {
  const parentRef = useRef(value)

  // sync the parentRef
  useLayoutEffect(() => {
    parentRef.current = value
  }, [value])

  // ...
      <ObjectInputField parentRef={parentRef} {/* ... */} />
  // ...
}
```

Reply:
oooo new idea!: could we utilize the path to grab the parent value? no need to pass the parent at all then

```js
import {get} from 'lodash'

// in the child
const parent = get(document, path.slice(0, path.length - 1))
```

Reply:
I like the path idea, but getting the parent this way is arguably less convenient than having it available on the `parent` context key? I think also we _do_ want to re-render each field's hidden check when the parent changes, and making it a ref here would potentially prevent fields from re-checking when the parent value changes, and therefore not update correctly? In practice, they will all update anyway since the document changes all of the time. As it is now, the re-rendering the ConditionalField component is cheap, so not sure if it's worth the added complexity.